core/src/providers/provider.rs

```diff
@@ -48,9 +51,10 @@ impl FromStr for ProviderID {
             "ai21" => Ok(ProviderID::AI21),
             "azure_openai" => Ok(ProviderID::AzureOpenAI),
             "anthropic" => Ok(ProviderID::Anthropic),
+            "mistralai" => Ok(ProviderID::MistralAI),
```

Comment:
This string is use visible and not great great. I would go with `mistral` or `mistral_ai` ?